example/home/mutations.py

```diff
@@ -0,0 +1,31 @@
+from wagtail.core import hooks
+import uuid
+from django.utils.crypto import get_random_string
+from wagtail.core.models import Page
+import graphene
+
+from grapple.types.pages import PageInterface
+from home.models import AuthorPage
+
+
+class CreateAuthor(graphene.Mutation):
+    class Arguments:
+        name = graphene.String()
+        parent = graphene.Int()
+
+    ok = graphene.Boolean()
+    author = graphene.Field(
+        PageInterface,
+    )
+
+    def mutate(root, info, name, parent):
+        # We use uuid here in order to ensure the slug will always be unique accross tests
+        author = AuthorPage(name=name, title=name, slug=uuid.uuid4().hex[:6].upper())
```

Comment:
This probably warrants a note in the docs that any custom mutation should take care of validation

Reply:
In fact, validation errors are properly handled. 
I explicitely added this randomly generator slug to avoid validation errors, but an existing slug would result in the response below
```json
{
  "errors": [
    {
      "message": "{'slug': ['This slug is already in use']}",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "createAuthor"
      ]
    }
  ],
  "data": {
    "createAuthor": null
  }
}
```